src/node_file.cc

```diff
@@ -138,9 +135,25 @@ void FSReqWrap::Resolve(Local<Value> value) {
   MakeCallback(env()->oncomplete_string(), arraysize(argv), argv);
 }
 
+void FSReqWrap::Init(const char* syscall,
+                     const char* data,
+                     enum encoding encoding,
+                     Ownership ownership) {
+  CHECK_EQ(info_, nullptr);  // Only initialize once.
+  const bool copy = (data != nullptr && ownership == Ownership::COPY);
+  const size_t size = copy ? 1 + strlen(data) : 0;
+  char* const storage = new char[sizeof(*info_) + size];
+  info_ = new(storage) FSReqInfo(syscall, data, encoding);
```

Comment:
It's arguably neater and less error prone to move this into FSReqInfo.

Reply:
The placement new logic.  I.e., make it so there is only one way to create an FSReqInfo instance and make that logic part of the class itself.